SDWebImageWebPCoder/Classes/SDImageWebPCoder.h

```diff
@@ -12,11 +12,29 @@
 @import SDWebImage;
 #endif
 
+#if __has_include("webp/decode.h") && __has_include("webp/encode.h") && __has_include("webp/demux.h") && __has_include("webp/mux.h")
+#import "webp/decode.h"
+#import "webp/encode.h"
+#import "webp/demux.h"
+#import "webp/mux.h"
+#elif __has_include(<libwebp/decode.h>) && __has_include(<libwebp/encode.h>) && __has_include(<libwebp/demux.h>) && __has_include(<libwebp/mux.h>)
+#import <libwebp/decode.h>
+#import <libwebp/encode.h>
+#import <libwebp/demux.h>
+#import <libwebp/mux.h>
+#else
+@import libwebp;
+#endif
+
 /**
  Built in coder that supports WebP and animated WebP
  */
 @interface SDImageWebPCoder : NSObject <SDProgressiveImageCoder, SDAnimatedImageCoder>
 
 @property (nonatomic, class, readonly, nonnull) SDImageWebPCoder *sharedCoder;
 
+- (void) updateWebPOptionsToConfig:(WebPConfig * _Nonnull)config
+                       maxFileSize:(NSUInteger)maxFileSize
+                           options:(nullable SDImageCoderOptions *)options;
+
```

Comment:
One question, should this API be exposed to public ?


Actually, this WebP Coder is a top-level wrapper. So I don't want user to touch `WebPConfig` low level struct or something.

Can this be marked as private ? Users can just use the public API from `SDImageCoder` protocol. Like `decodedImageWithData:options:`

Reply:
If you want to call this API in Unit Test. One tip:

```objective-c
// Put these code into SDWebImageWebPCoderTests.m 


@interfae SDImageWebPCoder ()
- (void) updateWebPOptionsToConfig:(WebPConfig * _Nonnull)config
                       maxFileSize:(NSUInteger)maxFileSize
                           options:(nullable SDImageCoderOptions *)options;
@end

````

Reply:
Good point! 

Does not need to be public, just added it there to be able to test it in unit tests. (My Obj-c skills are bit rusty as I have mainly done Swift development for past three years. 😄 ) I'll try to do it in the way you suggested!